packages/transactions/src/durable-nonce.ts

```diff
@@ -16,20 +20,137 @@ type AdvanceNonceAccountInstruction<
 }
+
+export function assertIsDurableNonceTransaction(
+    transaction: BaseTransaction | (BaseTransaction & IDurableNonceTransaction)
+): asserts transaction is BaseTransaction & IDurableNonceTransaction {
+    if (!isDurableNonceTransaction(transaction)) {
+        // TODO: Coded error.
+        throw new Error('Transaction is not a durable nonce transaction');
+    }
+}
+
+function createAdvanceNonceAccountInstruction<
+    TNonceAccountAddress extends string = string,
+    TNonceAuthorityAddress extends string = string
+>(
+    nonceAccountAddress: Base58EncodedAddress<TNonceAccountAddress>,
+    nonceAuthorityAddress: Base58EncodedAddress<TNonceAuthorityAddress>
+): AdvanceNonceAccountInstruction<TNonceAccountAddress, TNonceAuthorityAddress> {
+    return {
+        accounts: [
+            { address: nonceAccountAddress, role: AccountRole.WRITABLE },
+            {
+                address: RECENT_BLOCKHASHES_SYSVAR_ADDRESS,
+                role: AccountRole.READONLY,
+            },
+            { address: nonceAuthorityAddress, role: AccountRole.READONLY_SIGNER },
+        ],
+        data: new Uint8Array([4, 0, 0, 0]) as AdvanceNonceAccountInstructionData,
+        programAddress: SYSTEM_PROGRAM_ADDRESS,
+    };
+}
+
+function isAdvanceNonceAccountInstruction(instruction: IInstruction): instruction is AdvanceNonceAccountInstruction {
+    return (
+        instruction.programAddress === SYSTEM_PROGRAM_ADDRESS &&
+        // Test for `AdvanceNonceAccount` instruction data
+        instruction.data != null &&
+        isAdvanceNonceAccountInstructionData(instruction.data) &&
+        // Test for exactly 3 accounts
+        instruction.accounts?.length === 3 &&
+        // First account is nonce account address
+        instruction.accounts[0].address != null &&
+        instruction.accounts[0].role === AccountRole.WRITABLE &&
+        // Second account is recent blockhashes sysvar
+        instruction.accounts[1].address === RECENT_BLOCKHASHES_SYSVAR_ADDRESS &&
+        instruction.accounts[1].role === AccountRole.READONLY &&
+        // Third account is nonce authority account
+        instruction.accounts[2].address != null &&
+        instruction.accounts[2].role === AccountRole.READONLY_SIGNER
+    );
+}
+
+function isAdvanceNonceAccountInstructionData(data: Uint8Array): data is AdvanceNonceAccountInstructionData {
+    // AdvanceNonceAccount is the fifth instruction in the System Program (index 4)
+    return data.byteLength === 4 && data[0] === 4 && data[1] === 0 && data[2] === 0 && data[3] === 0;
+}
+
+function isDurableNonceTransaction(
+    transaction: BaseTransaction | (BaseTransaction & IDurableNonceTransaction)
+): transaction is BaseTransaction & IDurableNonceTransaction {
+    return (
+        'lifetimeConstraint' in transaction &&
+        typeof transaction.lifetimeConstraint.nonce === 'string' &&
+        transaction.instructions[0] != null &&
+        isAdvanceNonceAccountInstruction(transaction.instructions[0])
```

Comment:
I think this answers my one question.
This first instruction is always `AdvanceNonce` in a DNT ?

Reply:
Always!